### PR TITLE
Add fuctionality allowing for streaming audio from Youtube

### PR DESCRIPTION
**Pull request by szyjah**
requirements.txt file size difference makes my brain go OOF.

### CODE REVIEW

File: ytplayer.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            raise commands.NoPrivateMessage('This command can\'t be used in DM channels.')
+        return True
+    
+    async def cog_command_error(self, ctx: commands.Context, error: commands.CommandError):
+        await ctx.send("Nie mam ochoty działać poprawnie | "+ str(error))
+
+    @commands.command(name='play')
+    async def _jutub(self,ctx: commands.Context, *args):     
+        channel = ctx.message.author.voice.channel
+        if(len(self.bot.voice_clients) == 0 and channel != None):
```
Comment (Niewidzialny84) [inline ytplayer.py:32]: Check if sender is not a bot

File: ytplayer.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    @commands.command(name='resume')
+    async def _resume(self,ctx: commands.Context):
+        self.voice.resume()
+
+    @commands.command(name='stop')
+    async def _stop(self,ctx: commands.Context):
+        self.voice.stop()
+
+def setup(bot):
+    """Add component"""
```
Comment (Niewidzialny84) [inline ytplayer.py:62]: All commands should have a this comment to get auto documented in help command

**Review by Niewidzialny84 — CHANGES_REQUESTED** [review]
Probably requirements.txt will be toasted now but ok nothing to do here for you, fix the comments

**Comment by seker212** [comment]
@szyjah Have you tested what happens when you try the following?
1. Use `.sound` while bot is playing on the same channel
2. Use `.sound` while bot is playing on a different channel
3. Use `.sound` after `.pause` (and than for example use `.resume`

File: ytplayer.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    async def _pause(self,ctx: commands.Context):
+        self.voice.pause()
+
+    @commands.command(name='resume')
+    async def _resume(self,ctx: commands.Context):
+        self.voice.resume()
+
+    @commands.command(name='stop')
+    async def _stop(self,ctx: commands.Context):
+        self.voice.stop()
```
Comment (seker212) [inline ytplayer.py:59]: Shouldn't bot disconnect form channel after `stop` command?

File: ytplayer.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import urllib.request
+
+FFMPEG_OPTIONS = {'before_options': '-reconnect 1 -reconnect_streamed 1 -reconnect_delay_max 5','options': '-vn'}
+
+class YTPlayer(commands.Cog):
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.channel = None
+        self.voice = None
+        self.timezone = None
```
Comment (seker212) [inline ytplayer.py:16]: Not used anywhere

**Review by seker212 — CHANGES_REQUESTED** [review]

File: ytplayer.py (hunk 4 above)
Comment (Niewidzialny84) [reply ytplayer.py:16]: `self.voice` is actually used

Comment (szyjah) [reply ytplayer.py]: Unused variables have been deleted.

**Review by Niewidzialny84 — APPROVED** [review]
I think yes

**Review by seker212 — APPROVED** [review]
I vote for YES

**Pull request merged**